internal/controller/promotionstrategy_controller.go

```diff
@@ -164,11 +164,14 @@ func (r *PromotionStrategyReconciler) upsertChangeTransferPolicy(ctx context.Con
 		},
 	}
 
-	previousEnvironmentCommitStatusSelector := promoterv1alpha1.CommitStatusSelector{
-		Key: promoterv1alpha1.PreviousEnvironmentCommitStatusKey,
-	}
-	if !slices.Contains(pcNew.Spec.ProposedCommitStatuses, previousEnvironmentCommitStatusSelector) {
-		pcNew.Spec.ProposedCommitStatuses = append(pcNew.Spec.ProposedCommitStatuses, previousEnvironmentCommitStatusSelector)
+	previousEnvironmentIndex, _ := utils.GetPreviousEnvironmentStatusByBranch(*ps, environment.Branch)
+	if previousEnvironmentIndex > 0 && (len(ps.Spec.ActiveCommitStatuses) > 0 || len(ps.Spec.Environments[previousEnvironmentIndex].ActiveCommitStatuses) > 0) {
```

Comment:
Should this be `previousEnvironmentIndex >= 0`?

Reply:
No, because we don't want to create it on the lowest level environment either

Reply:
But if the previous environment's index is `0`, doesn't that mean we're currently working on environment `1`?